processor/src/main/java/org/mapstruct/ap/internal/model/assignment/ExistingInstanceSetterWrapperForCollectionsAndMaps.java

```diff
@@ -47,7 +50,7 @@ public ExistingInstanceSetterWrapperForCollectionsAndMaps(Assignment decoratedAs
             fieldAssignment,
             mapNullToDefault
         );
-        this.includeSourceNullCheck = ALWAYS == nvms;
+        this.includeSourceNullCheck = ALWAYS == nvms || IGNORE == nvpms;
```

Comment:
Shouldn't the source null check be independent of the `NullValuePropertyMappingStrategy`. We should not do a mistake again and have usages of some properties affecting others

Reply:
I don't think the naming of this is very appropriate. The property takes care of including the else branch. I'll check it out.